### DISCUSSION TITLE
ESP32

### DISCUSSION THREAD

**Discussion by J-GitHub1** [Winston-Lu/ESP8266-LED-Shelf#11] [Q&A] [answered]
Great project!

I have a load of ESP32, I cannot seem to get this project to work on one.

I feel like this should be simple enough, but I've tried really hard and I cannot make it work.

Any pointers?

**Comment by Winston-Lu** [comment]
I believe you just need to replace the ESP8266 specific libraries with an ESP32 counterpart. So far it seems that [NTPTime.cpp](https://github.com/Winston-Lu/ESP8266-LED-Shelf/blob/master/NTPTime.cpp) and [WebServer.cpp](https://github.com/Winston-Lu/ESP8266-LED-Shelf/blob/master/WebServer.cpp) are the only files affected when you switch to the ESP32 board. You could try using [this library from expressif](https://github.com/espressif/arduino-esp32/blob/master/libraries/WebServer/src/WebServer.h) in its place, but i'm not sure if you will come up with any incompatibilities. I can also look into it but as of now I don't have as much time to work on it

**Reply by J-GitHub1** [reply, marked as answer]
I don't want to mess up anything on Github; I've forked the project and created a separate branch for the platformIO version. I've also changed one of the required libraries and added it to: https://github.com/J-GitHub1/ESPAsyncWebServer, no idea how to add this in Arduino IDE, though. 

The benefit of using platformIO is that it has a file system build and filesystem upload built in, but it takes a bit of a learning curve. When I have more time, I'll rewrite the READme for PlatformIO. 
 
I followed the wiring of [florianL21](https://github.com/florianL21/LED-ClockShelf); I used the Blynk Setup, but the Blynk App on iOS is rubbish. 

Amazing work Winston-Lu, and thanks for the clear instruct

![IMG_1930](https://github.com/Winston-Lu/ESP8266-LED-Shelf/assets/104653841/f94b1c0f-76a9-4017-9d91-d3a1043c50f3)
ions. https://github.com/florianL21/LED-ClockShelf


![IMG_1926](https://github.com/Winston-Lu/ESP8266-LED-Shelf/assets/104653841/951699ab-292f-41d0-a8e8-472aff848f90)

I've said this a dozen times I have no idea what I'm doing, so if any of this is wrong, I'm more than happy to take feedback.

**Reply by Winston-Lu** [reply]
Looks great, I'll just add a link to your project since our build systems aren't quite the same. Having a different branch requiring a different IDE and build tools doesn't seem reasonable, but I'll happily promote your repo on mine. You may also want to put the PlatformIO branch as the default branch so they know which one to use
